core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -104,13 +105,14 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
   /**
    * Maps the error value of this effect to the specified constant value.
    */
-  final def asError[E1](e1: => E1): ZIO[R, E1, A] = mapError(new ZIO.ConstFn(() => e1))
+  final def asError[E1](e1: => E1)(implicit ev: CanFail[E]): ZIO[R, E1, A] =
```

Comment:
This one makes sense because for an effect with `E = Nothing`, it can already be treated as a subtype of a more specific error type.

Reply:
Okay.